.gitignore

```diff
@@ -7,5 +7,5 @@
 /doc
 /pkg
 /tags
-/Gemfile.lock
```

Comment:
for gems, it is conventional to ignore the Gemfile.lock because the `.gemspec` should describe depdencies

Reply:
so if there are specific versions of dependencies needed to work, we should make the `.gemspec` more specific (usually via `>=`, `<` requirements)

Reply:
TIL

Reply:
On researching this further, `bundler` docs recommend that you should check in `Gemfile.lock`, see https://bundler.io/guides/faq.html#using-gemfiles-inside-gems.

Further, as this is only for our consumption and not a true gem (in the sense that we don't publish it to `rubygems.org` for others), we should check in `Gemfile.lock` using the same principle we use for other (non-gem) repos.

Reply:
I stand corrected, thanks for finding that! Maybe the advice has changed since I last checked

Reply:
Indeed it has changed. See https://bundler.io/v2.5/man/bundle-install.1.html#THE-GEMFILE-LOCK.
> When Bundler first shipped, the Gemfile.lock was included in the .gitignore file included with generated gems. Over time, however, it became clear that this practice forces the pain of broken dependencies onto new contributors, while leaving existing contributors potentially unaware of the problem. Since bundle install is usually the first step towards a contribution, the pain of broken dependencies would discourage new contributors from contributing. As a result, we have revised our guidance for gem authors to now recommend checking in the lock for gems.